src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -45,14 +49,89 @@ static Startup()
             {
-                BuildChannel entity = entry.Entity;
                 DbContext context = entry.Context;
+                BuildChannel entity = entry.Entity;
+
+                StartAssociatedReleasePipelinesAsync(entry).GetAwaiter().GetResult();
+
                 var queue = context.GetService<BackgroundQueue>();
                 var dependencyUpdater = context.GetService<IDependencyUpdater>();
                 queue.Post(() => dependencyUpdater.StartUpdateDependenciesAsync(entity.BuildId, entity.ChannelId));
             };
         }
 
+        private static async Task StartAssociatedReleasePipelinesAsync(IInsertedEntry<BuildChannel, DbContext> entry)
+        {
+            BuildAssetRegistryContext context = entry.Context as BuildAssetRegistryContext;
+            ILogger<BuildChannel> logger = context.GetService<ILogger<BuildChannel>>();
+            BuildChannel entity = entry.Entity;
+            Build barBuildInfo = entity.Build;
+            Channel channel = context.Channels
+                .Where(ch => ch.Id == entity.ChannelId)
+                .Include(ch => ch.ChannelReleasePipelines)
+                .ThenInclude(crp => crp.ReleasePipeline)
+                .First();
+
+            // If something use the old API version we won't have this information available.
+            // This will also be the case if something adds an existing build (created using 
+            // the old API version) to a channel
+            if (barBuildInfo.AzureDevOpsBuildId == null)
+            {
+                logger.LogWarning($"barBuildInfo.AzureDevOpsBuildId is null for BAR Build.Id {barBuildInfo.Id}.");
+                return;
+            }
+
+            if (channel.ChannelReleasePipelines?.Any() != true)
+            {
+                logger.LogWarning($"Channel {channel.Id}, which build with BAR ID {barBuildInfo.Id} is attached to, doesn't have an associated publishing pipeline.");
+                return;
+            }
+
+            var azdoTokenProvider = context.GetService<IAzureDevOpsTokenProvider>();
+            var accessToken = azdoTokenProvider.GetTokenForAccount(barBuildInfo.AzureDevOpsAccount).GetAwaiter().GetResult();
+
+            var azdoClient = new AzureDevOpsClient(accessToken, logger, null);
+
+            var azdoBuild = await azdoClient.GetBuildAsync(
+                barBuildInfo.AzureDevOpsAccount,
+                barBuildInfo.AzureDevOpsProject,
+                barBuildInfo.AzureDevOpsBuildId.Value);
+
+            foreach (var pipeline in channel.ChannelReleasePipelines)
+            {
+                try
+                {
+                    var account = pipeline.ReleasePipeline.Organization;
+                    var project = pipeline.ReleasePipeline.Project;
+                    var pipelineId = pipeline.ReleasePipeline.PipelineIdentifier;
+
+                    var pipeDef = await azdoClient.GetReleaseDefinitionAsync(account, project, pipelineId);
```

Comment:
Are we going to run into concurrency issues if multiple runs of the release pipeline happen at the same time? Does this segment need to lock?

Reply:
That's possible, although seems the chance of it happening are very low. @riarenas is working on an improvement on how we trigger the pipeline that will fix this problem as well: https://github.com/dotnet/arcade/issues/1994